source/speech/speech.py

```diff
@@ -1149,7 +1164,11 @@ def speakTextInfo(
 
 	speechGen = GeneratorWithReturn(speechGen)
 	for seq in speechGen:
-		speak(seq, priority=priority)
+		speak(
+			seq,
+			priority=priority,
+			suppressBlanks=suppressBlanks or reason == OutputReason.SAYALL

```

Comment:
Why is there an exception for say all? Can you add a comment to explain.

Reply:
> Why is there an exception for say all? Can you add a comment to explain.

This is consistent with logic in `speech.speech.getTextInfoSpeech`, where "blank" is added to the speech sequence only if it isn't a sayall. As I understand, blanks are not meant to be spoken during a sayall and that this was something generally known, so not requiring a comment.

snippet from `speech.speech.getTextInfoSpeech`:

```
if not suppressBlanks and reason != OutputReason.SAYALL and shouldConsiderTextInfoBlank:
	# Translators: This is spoken when the line is considered blank.
	speechSequence.append(_("blank"))
```

Also, just before asking about timeline on review, I added a comment pointing out that comparing `reason == OutputReason.SAYALL` is redundant since in `speech\__init__.py`, when SayAll is initialized, it is passed a modified speech function where `suppressBlanks` is overridden to be true. So, I was thinking of removing all these comparisns like this, since sayall is already handled directly at the source. But, I wanted to get feedback on that first. What do you think?

Reply:
>As I understand, blanks are not meant to be spoken during a sayall and that this was something generally known, so not requiring a comment.

I'd prefer a comment. It's very easy for priors to change, reasoning helps promoting questioning the behavior. I.E. does this reason still exist / make sense in the current context. There is also the aspect of making it easy for developers with different histories to get up to speed on the reasoning.

>So, I was thinking of removing all these comparisns like this, since sayall is already handled directly at the source. But, I wanted to get feedback on that first. What do you think?

Yes, I think that would be preferable.